core/store/src/db.rs

```diff
@@ -550,35 +680,62 @@ impl RocksDB {
     }
 
     fn new_read_only<P: AsRef<std::path::Path>>(path: P) -> Result<Self, DBError> {
-        let options = Options::default();
-        let cf_names: Vec<_> = vec!["col0".to_string()];
-        let db = DB::open_cf_for_read_only(&options, path, cf_names.iter(), false)?;
-        let cfs =
-            cf_names.iter().map(|n| db.cf_handle(n).unwrap() as *const ColumnFamily).collect();
-        Ok(Self { db, cfs, _pin: PhantomPinned })
+        RocksDBOptions::default().read_only(path)
     }
 
     pub fn new<P: AsRef<std::path::Path>>(path: P) -> Result<Self, DBError> {
-        use strum::IntoEnumIterator;
-        let options = rocksdb_options();
-        let cf_names: Vec<_> = DBCol::iter().map(|col| format!("col{}", col as usize)).collect();
-        let cf_descriptors = DBCol::iter().map(|col| {
-            ColumnFamilyDescriptor::new(format!("col{}", col as usize), rocksdb_column_options(col))
-        });
-        let db = DB::open_cf_descriptors(&options, path, cf_descriptors)?;
-        #[cfg(feature = "single_thread_rocksdb")]
-        {
-            // These have to be set after open db
-            let mut env = Env::default().unwrap();
-            env.set_bottom_priority_background_threads(0);
-            env.set_high_priority_background_threads(0);
-            env.set_low_priority_background_threads(0);
-            env.set_background_threads(0);
-            println!("Disabled all background threads in rocksdb");
+        RocksDBOptions::default().read_write(path)
+    }
+
+    /// Checks if there is enough memory left to perform a write. Not having enough memory left can
+    /// lead to difficult to recover from state, thus a PreWriteCheckErr is pretty much
+    /// unrecoverable in most cases.
+    fn pre_write_check(&self) -> Result<(), PreWriteCheckErr> {
+        let counter = self.check_free_space_counter.fetch_add(1, Ordering::Relaxed);
+        if self.check_free_space_interval <= counter {
+            return Ok(());
         }
-        let cfs =
-            cf_names.iter().map(|n| db.cf_handle(n).unwrap() as *const ColumnFamily).collect();
-        Ok(Self { db, cfs, _pin: PhantomPinned })
+        self.check_free_space_counter.swap(0, Ordering::Relaxed);
+
+        let available = available_space(self.db.path())?;
+
+        if available < 16_u64 * self.free_space_threshold {
+            warn!("remaining disk space is running low ({} left)", available);
+        }
+
+        if available < self.free_space_threshold {
+            Err(PreWriteCheckErr::LowMemory(available))
```

Comment:
I thin this should be named LowDiskSpace. Low memory -> Low RAM

Reply:
Agree!